src/DataTransferObjectError.php

```diff
@@ -8,11 +8,23 @@
 
 class DataTransferObjectError extends TypeError
 {
+    use HasErrorProperties;
+
+    public function __construct($message, $properties = [])
+    {
+        parent::__construct($message);
+        $this->setProperties($properties);
+    }
+
     public static function unknownProperties(array $properties, string $className): DataTransferObjectError
     {
         $propertyNames = implode('`, `', $properties);
 
-        return new self("Public properties `{$propertyNames}` not found on {$className}");
+        return new self("Public properties `{$propertyNames}` not found on {$className}", [
+            'error'      => __FUNCTION__,
```

Comment:
🤔
Wouldn't it make more sense to have dedicated exception for all this (for backwards compat extend from `DataTransferObjectError`) ?

Reply:
I definitely think that's the best approach as well. See [this comment](https://github.com/spatie/data-transfer-object/pull/96#issuecomment-595150363) - I'm going to refactor my changes to add separate exceptions for each error type.